File: 2133-check-if-every-row-and-column-contains-all-numbers/2133-check-if-every-row-and-column-contains-all-numbers.java
class Solution {
   public boolean checkValid(int[][] matrix) {
			int rowsLength = matrix.length;
			int colsLength = matrix[0].length;
			HashSet<String> elements = new HashSet<>();
			for(int i=0;i<rowsLength;i++){
				for(int j=0;j<colsLength;j++){
					if(!elements.add("Row"+i+"ele"+matrix[i][j]) || !elements.add("Col"+j+"ele"+matrix[i][j]))
						return false;
				}
			}
			return true;
    }
}

File: 2133-check-if-every-row-and-column-contains-all-numbers/2133-check-if-every-row-and-column-contains-all-numbers.cpp
class Solution {
public:
    bool checkValid(vector<vector<int>>& matrix) {
        int row=matrix.size();
        int col=matrix[0].size();
        for(int i=0;i<row;i++){
            set<int>temp;
            for(int j=0;j<col;j++){
                temp.insert(matrix[i][j]);
            }
            if(temp.size()!=row)
                return false;
        }
        for(int i=0;i<col;i++){
            set<int>temp;
            for(int j=0;j<row;j++){
                temp.insert(matrix[j][i]);
            }
            if(temp.size()!=row)
                return false;
        }
        return true;
    }
};